EKF/control.cpp

```diff
@@ -674,7 +673,31 @@ void Ekf::controlGpsFusion()
 			// We haven't had an absolute position fix for a longer time so need to do something
 			do_reset = do_reset || ((_time_last_imu - _time_last_pos_fuse) > (2 * _params.reset_timeout_max));
 
-			if (do_reset) {
+			// A reset to the EKF-GSF estimate can be performed after a recent takeoff which will enable
+			// recovery from a bad magnetometer or field estimate.
+			// This special case reset can also be requested externally.
+			// The minimum time interval between resets to the EKF-GSF estimate must be limited to
+			// allow the EKF-GSF time to improve its estimate if the first reset was not successful.
+			const bool stopped_following_gps_velocity = _time_last_imu > (_params.EKFGSF_reset_delay + _time_last_vel_fuse) &&
+							   (_time_last_vel_fuse > _time_last_on_ground_us);
+			const bool recent_takeoff = (_control_status.flags.in_air && (_imu_sample_delayed.time_us - _time_last_on_ground_us) < 30000000);
```

Comment:
There is one subtraction of two unsigned integers left here.

Reply:
See https://github.com/PX4/ecl/pull/751/commits/a27635e18420763481439e0a0b14e28dd9377198